src/Utilities/Compiler/SmallDictionary.cs

```diff
@@ -137,13 +137,13 @@ public void Remove(K key)
 
             if (balance == -2)
             {
-                rotated = currentNode.Right!.Balance < 0 ?
+                rotated = currentNode.Right!.Balance <= 0 ?
```

Comment:
This utility type has been cloned from Roslyn: https://github.com/dotnet/roslyn/blob/6146f0e9a5435cf0bfb41d3419d665b7f1a68a84/src/Compilers/Core/Portable/Collections/SmallDictionary.cs#L339

Would be good to get more eyes here. Tagging @sharwell.

Reply:
Right, most part copied from there, but if did not had a Remove functoriality, I added this later and this part is pretty much copied from Insert function, though in Remove case the Balance 0, which missed here